lib/vintage_net_wizard/web/api.ex

```diff
@@ -0,0 +1,88 @@
+defmodule VintageNetWizard.Web.Api do
+  @moduledoc false
+
+  alias VintageNetWizard.{WiFiConfiguration, Backend}
+  alias Plug.Conn
+
+  use Plug.Router
+
+  plug(Plug.Parsers, parsers: [:json], json_decoder: Jason)
+  plug(:match)
+  plug(:dispatch)
+
+  get "/access_points" do
+    {:ok, access_points} =
+      Backend.access_points()
+      |> to_json()
+
+    send_json(conn, 200, access_points)
+  end
+
+  get "/configurations" do
+    {:ok, json} =
+      Backend.configurations()
+      |> Jason.encode()
+
+    send_json(conn, 200, json)
+  end
+
+  put "/configurations" do
+    :ok =
+      conn
+      |> get_body()
+      |> Enum.map(fn cfg ->
+        {:ok, cfg} = WiFiConfiguration.from_map(cfg)
+        cfg
+      end)
+      |> Backend.save()
+
+    send_json(conn, 204, "")
+  end
+
+  post "/apply" do
+    :ok = Backend.apply()
+    send_json(conn, 202, "")
+  end
+
+  defp send_json(conn, status_code, json) do
+    conn
+    |> put_resp_content_type("application/json")
+    |> send_resp(status_code, json)
+  end
+
+  defp get_body(%Conn{body_params: %{"_json" => body}}) do
+    body
+  end
+
+  defp get_body(%Conn{body_params: body}), do: body
+
+  defp to_json(access_points) do
+    access_points
+    |> Enum.map(fn {_, ap} -> summarize(ap) end)
+    |> Enum.filter(&non_empty_ssid/1)
+    |> Enum.sort(fn %{signal_percent: a}, %{signal_percent: b} -> a >= b end)
+    |> Enum.uniq_by(fn %{ssid: ssid} -> ssid end)
+    |> Jason.encode()
+  end
+
+  defp summarize(%{
```

Comment:
I PR'd vintage_net to change the AP reports: https://github.com/nerves-networking/vintage_net/pull/95. After that's in, how about dropping the whole `summarize` thing and sending the whole map out. That adds `bssid` and `signal_dbm`, and that seems harmless.

Reply:
I am good with that, once that is in and we upgrade I can update and fix the JSON API spec too.